bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveIndicatorCommandClass.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zwave.internal.protocol.commandclass;
+
+import org.openhab.binding.zwave.internal.config.ZWaveDbCommandClass;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage;
+import org.openhab.binding.zwave.internal.protocol.ZWaveController;
+import org.openhab.binding.zwave.internal.protocol.ZWaveEndpoint;
+import org.openhab.binding.zwave.internal.protocol.ZWaveNode;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage.SerialMessageClass;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage.SerialMessagePriority;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage.SerialMessageType;
+import org.openhab.binding.zwave.internal.protocol.event.ZWaveCommandClassValueEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+
+/**
+ * Handles the Indicator command class.
+ * The indicator command class operates the indicator on the physical device if available. 
+ * This can be used to identify a device or use the indicator for special purposes.
+ * Example is the Evolve LCD panel that uses the Indicator class to toggle the labels 
+ * displayed on the LCD. The Indicator class is also used to sync multiple panels' labels
+ * 
+ * @author Pedro Paixao
+ * @since 1.8.0
+ */
+
+@XStreamAlias("indicatorCommandClass")
+public class ZWaveIndicatorCommandClass extends ZWaveCommandClass implements ZWaveIndicatorCommands {
+
+	@XStreamOmitField
+	private static final Logger logger = LoggerFactory.getLogger(ZWaveIndicatorCommandClass.class);
+	
+	private static final int INDICATOR_SET = 0x01;
+	private static final int INDICATOR_GET = 0x02;
+	private static final int INDICATOR_REPORT = 0x03;
+
+	private boolean isGetSupported = true;
+
+	/**
+	 * Creates a new instance of the ZWaveIndicatorCommandClass class.
+	 * @param node the node this command class belongs to
+	 * @param controller the controller to use
+	 * @param endpoint the endpoint this Command class belongs to
+	 */
+	public ZWaveIndicatorCommandClass(ZWaveNode node,
+			ZWaveController controller, ZWaveEndpoint endpoint) {
+		super(node, controller, endpoint);
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public CommandClass getCommandClass() {
+		return CommandClass.INDICATOR;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void handleApplicationCommandRequest(SerialMessage serialMessage,
+			int offset, int endpoint) {
+		logger.debug("NODE {}: Received Indicator Request", this.getNode().getNodeId());
+		int command = serialMessage.getMessagePayloadByte(offset);
+		switch (command) {
+			case INDICATOR_SET:
```

Comment:
Just to be clear - this function is for receiving messages sent FROM the device - and as per my previous comments, I don't think the device will send a SET command and this should be removed unless you confirm that the device is sending SET commands.